ca_candidates/people.py

```diff
@@ -0,0 +1,205 @@
+                    other = other.xpath('./ul/li/a/@href')
+                    for contact in other:
+                        if "mailto:" in contact:
+                            email = contact.replace("mailto:", "")
+                            p.add_contact("email", email)
+                        else:
+                            p.add_link(contact)
+                else: 
+                    for contact in contacts:
+                        if "mailto:" in contact:
+                            email = contact.replace("mailto:", "")
+                            p.add_contact("email", email)
+                        else:
+                            p.add_link(contact)
+
+            yield p
+
+    
+    def scrape_liberal(self):
+        page = self.lxmlize(LIBERAL_PAGE)
+
+        candidates = page.xpath('//div[@class="person-listing-container"]/article')
+        assert(len(candidates))
+        
+        for candidate in candidates:
+            name = candidate.xpath('./div/header/h2/text()')[0]
+            district = candidate.xpath('./div/header/h3/text()')[0]
+            image = candidate.xpath('./div/div')[0]
+
+            p = Person(primary_org="lower", name=name, district=district, role="candidate", party="Liberal Party")
+            # , image=image)
+            # image is still a div element -> extract url
+
+            p.add_source(LIBERAL_PAGE)
+
+            row = candidate.xpath('./div/div')[1]
+            contacts = row.xpath('./div/a/@href')
+            for contact in contacts:
+                if "facebook.com" in contact or "twitter.com" in contact or "instagram.com" in contact or "x.com" in contact or "linkedin.com" in contact or "youtube.com" in contact:
+                    p.add_link(contact)
+                else: # candidate may have individual page with more information --> TO BE DONE
+                    candidatepage = self.lxmlize(contact)
+                    info = candidatepage.xpath('//section[@class="sidebar2-section"]/div/a/@href')
+                    email = candidatepage.xpath('//a[contains(@href, "mailto:")]/@href')
+                    # email = candidatepage.xpath('//a[contains(@href, "mailto:")]//text()')
+                    if not email == []:
+                        email = email[0].replace("mailto:", "")
+                        if email == 'info@johngoheenliberal.ca?subject=I%20want%20to%20volunteer': # change logic here
+                            email = 'info@johngoheenliberal.ca'
+                        elif email == 'connect@wadechang.ca?subject=Hi%20Wade%2C%20this%20is%20...':
+                            email = 'connect@wadechang.ca'
+                        email = email.replace("Canada￼", "")
+                        p.add_contact("email", email)
+                    p.add_source(contact)
+                    
+            yield p
+    
+    def scrape_green(self):
+        page1 = self.lxmlize(GREEN_PARTY_PAGE + "page/1")
```

Comment:
So this will need to potentially change @zilinweng as we have just shifted to making this an AJAX based page.